ukat/segment/whole_kidney.py

```diff
@@ -0,0 +1,207 @@
+        Returns
+        -------
+        tkv : float
+            Total kidney volume
+        """
+        return self.volumes['total']
+
+    def get_lkv(self):
+        """
+        Get the left kidney volume in milliliters.
+        Returns
+        -------
+        lkv : float
+            Left kidney volume
+        """
+        return self.volumes['left']
+
+    def get_rkv(self):
+        """
+        Get the right kidney volume in milliliters.
+        Returns
+        -------
+        rkv : float
+            Right kidney volume
+        """
+        return self.volumes['right']
+
+    def to_nifti(self, output_directory=os.getcwd(), base_file_name='Output',
+                 maps='all'):
+        """Exports masks to NIFTI.
+
+        Parameters
+        ----------
+        output_directory : string, optional
+            Path to the folder where the NIFTI files will be saved.
+        base_file_name : string, optional
+            Filename of the resulting NIFTI. This code appends the extension.
+            Eg., base_file_name = 'Output' will result in 'Output.nii.gz'.
+        maps : list or 'all', optional
+            List of maps to save to NIFTI. This should either the string "all"
+            or a list of maps from ["mask", "left", "right", "individual"]
+        """
+        os.makedirs(output_directory, exist_ok=True)
+        base_path = os.path.join(output_directory, base_file_name)
+        if maps == 'all' or maps == ['all']:
+            maps = ['mask', 'left', 'right', 'individual']
+
+        if isinstance(maps, list):
+            for result in maps:
+                if result == 'mask':
+                    mask_nifti = nib.Nifti1Image(self._mask, self.affine)
+                    nib.save(mask_nifti, base_path + '_mask.nii.gz')
+                elif result == 'left':
+                    left_nifti = nib.Nifti1Image(self.get_left_kidney(),
+                                                 self.affine)
+                    nib.save(left_nifti, base_path + '_left_kidney.nii.gz')
+                elif result == 'right':
+                    right_nifti = nib.Nifti1Image(self.get_right_kidney(),
+                                                  self.affine)
+                    nib.save(right_nifti, base_path + '_right_kidney.nii.gz')
+                elif result == 'individual':
+                    ind_nifti = nib.Nifti1Image(self.get_kidneys(),
+                                                self.affine)
+                    nib.save(ind_nifti, base_path +
+                             '_individual_kidneys.nii.gz')
+        else:
+            raise ValueError('No NIFTI file saved. The variable "maps" '
+                             'should be "all" or a list of maps from '
+                             '"["mask", "left", "right", "individual"]".')
+
+    def _calculate_volume(self, mask):
+        mask = mask > 0.5
+        return np.sum(mask) * np.prod(self.header.get_zooms()) / 1000
```

Comment:
If you used self._nifti.header, would you still need `super().__init__(pixel_array, affine)` when initiating the class?

Reply:
Using super means all the normal nibabel image functions like `get_fdata()` still work.